ansible/roles/test/files/ptftests/fg_ecmp_test.py

```diff
@@ -0,0 +1,401 @@
+                if port_idx in active_port_grp:
+                    assert port_idx == port
+                    flow_redistribution_in_correct_grp = True
+                elif port_idx == self.first_nh:
+                    flow_redistribution_in_correct_grp = True
+                    tuple_to_port_map[src_ip] = port_idx
+                assert flow_redistribution_in_correct_grp == True
+
+            self.test_balancing(hit_count_map)
+            return
+
+        else:
+            self.log("Unsupported testcase " + self.test_case)
+            return
+
+
+    def send_rcv_ip_pkt(self, in_port, sport, dport, src_ip_addr, dst_ip_addr,
+                       dst_port_list, ipv4=True):
+        if ipv4:
+            (matched_index, received) = self.send_rcv_ipv4_pkt(in_port, sport, dport,
+                    src_ip_addr, dst_ip_addr, dst_port_list)
+        else:
+            (matched_index, received) = self.send_rcv_ipv6_pkt(in_port, sport, dport,
+                    src_ip_addr, dst_ip_addr, dst_port_list)
+
+        assert received
+
+        matched_port = dst_port_list[matched_index]
+        logging.info("Received packet at " + str(matched_port))
+
+        return (matched_port, received)
+
+
+    def send_rcv_ipv4_pkt(self, in_port, sport, dport,
+                         ip_src, ip_dst, dst_port_list):
+        src_mac = self.dataplane.get_mac(0, in_port)
+        rand_int = random.randint(1, 254)
+
+        pkt = simple_tcp_packet(
+                            eth_dst=self.router_mac,
+                            eth_src=src_mac,
+                            ip_src=ip_src,
+                            ip_dst=ip_dst,
+                            tcp_sport=sport,
+                            tcp_dport=dport,
+                            ip_ttl=64)
+        if self.inner_hashing:
+            pkt = simple_vxlan_packet(
+                    eth_dst=self.router_mac,
+                    eth_src=src_mac,
+                    ip_id=0,
+                    ip_src='2.2.2.' + str(rand_int),
+                    ip_dst=self.dst_ip,
+                    ip_ttl=64,
+                    udp_sport=rand_int,
+                    udp_dport=4789,
+                    vxlan_vni=rand_int,
+                    with_udp_chksum=False,
+                    inner_frame=pkt)
+
+        send_packet(self, in_port, pkt)
+
+        masked_exp_pkt = Mask(pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "ttl")
+
+        return verify_packet_any_port(self, masked_exp_pkt, dst_port_list)
+
+
+    def send_rcv_ipv6_pkt(self, in_port, sport, dport,
+                         ip_src, ip_dst, dst_port_list):
+        src_mac = self.dataplane.get_mac(0, in_port)
+        rand_int = random.randint(1, 254)
+
+        pkt = simple_tcp_packet(
```

Comment:
@anish-n IMHO, it will fail in case of IPv6 addresses.

Reply:
Fixed, thanks